Dockerfile

```diff
@@ -13,6 +17,12 @@ RUN bundle install --jobs 4 && npm install && npm install -g phantomjs-prebuilt
 
 EXPOSE 3000
 ADD . /myapp
-RUN RAILS_ENV=production bundle exec rake assets:precompile
+RUN if [ -z "$CUSTOM_ASSETS_URL" ]; then\
```

Comment:
About running the `download_assets` outside the container, I can do that. But if we're going to run the `assets:precompile` task inside, I still would need to set build arg `CUSTOM_ASSETS_PATHS` since assets:precompile needs it. 

Re: using the ADD directive. The `custom_assets:download` task is a little more complex than just download and untar. That's why I think it does need some kind of separate script. On top of that, you'd still need to pass the URL from where you're downloading the assets as a build-arg.